File: CS/WindowsFormsApplication8/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraCharts;

namespace WindowsFormsApplication8 {
    public partial class Form1 : Form {

        #region Constants & Variables
        static ImageList imgList = new ImageList();

        static string[] seriesArguments = new string[] { 
            "FAR WEST", "GREAT LAKES", "MIDEAST", "NEW ENGLAND", 
            "PLAINS", "ROCKY MOUNTAIN", "SOUTHEAST", "SOUTHWEST" };

        static Color[] seriesColors = new Color[] { 
            Color.Aqua, Color.Aquamarine, Color.BlueViolet, Color.Crimson, 
            Color.DarkOrange, Color.DarkSeaGreen, Color.Green, Color.Indigo };
        #endregion


        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {

            this.gSPTableAdapter.Fill(this.gspDataSet.GSP);
            imgList.Images.AddRange(CreateImages(seriesColors));
            pivotGridControl1.ValueImages = imgList;

        }


        private void pivotGridControl1_FieldValueImageIndex(object sender, PivotFieldImageIndexEventArgs e) {
            if (e.Field == fieldRegion && e.ValueType == PivotGridValueType.Value)
                e.ImageIndex = GetIndexByName(e.Value);
        }

        private void chartControl1_CustomDrawSeries(object sender, CustomDrawSeriesEventArgs e) {

            UpdateDrawOptions((BarDrawOptions)e.SeriesDrawOptions);
            UpdateDrawOptions((BarDrawOptions)e.LegendDrawOptions);

        }


        public static void UpdateDrawOptions(BarDrawOptions options) {

            options.FillStyle.FillMode = FillMode.Hatch;
            options.Shadow.Color = Color.Azure;            
        }


        #region Auxilliary Methods
        private Image[] CreateImages(Color[] colors) {
            Image[] images = new Image[colors.Length];
            for (int i = 0; i < colors.Length; i++) {
                images[i] = CreateImage(16, 16, colors[i]);
            }

            return images;
        }

        private Image CreateImage(int width, int height, Color imageColor) {
            Bitmap image = new Bitmap(width, height);

            using (Graphics gfx = Graphics.FromImage(image))
            using (SolidBrush brush = new SolidBrush(imageColor)) {
                gfx.FillRectangle(brush, 0, 0, width, height);
            }

            return (Image)image;
        }

        private static int GetIndexByName(object name) {
            if (name != null)
                for (int i = 0; i < seriesArguments.Length; i++)
                    if (seriesArguments[i] == name.ToString()) return i;

            return -1;
        }
        #endregion

    }
}

File: VB/WindowsFormsApplication8/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraCharts

Namespace WindowsFormsApplication8

    Public Partial Class Form1
        Inherits Form

'#Region "Constants & Variables"
        Private Shared imgList As ImageList = New ImageList()

        Private Shared seriesArguments As String() = New String() {"FAR WEST", "GREAT LAKES", "MIDEAST", "NEW ENGLAND", "PLAINS", "ROCKY MOUNTAIN", "SOUTHEAST", "SOUTHWEST"}

        Private Shared seriesColors As Color() = New Color() {Color.Aqua, Color.Aquamarine, Color.BlueViolet, Color.Crimson, Color.DarkOrange, Color.DarkSeaGreen, Color.Green, Color.Indigo}

'#End Region
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gSPTableAdapter.Fill(gspDataSet.GSP)
            imgList.Images.AddRange(CreateImages(seriesColors))
            pivotGridControl1.ValueImages = imgList
        End Sub

        Private Sub pivotGridControl1_FieldValueImageIndex(ByVal sender As Object, ByVal e As PivotFieldImageIndexEventArgs)
            If e.Field Is fieldRegion AndAlso e.ValueType = PivotGridValueType.Value Then e.ImageIndex = GetIndexByName(e.Value)
        End Sub

        Private Sub chartControl1_CustomDrawSeries(ByVal sender As Object, ByVal e As CustomDrawSeriesEventArgs)
            Call UpdateDrawOptions(CType(e.SeriesDrawOptions, BarDrawOptions))
            Call UpdateDrawOptions(CType(e.LegendDrawOptions, BarDrawOptions))
        End Sub

        Public Shared Sub UpdateDrawOptions(ByVal options As BarDrawOptions)
            options.FillStyle.FillMode = FillMode.Hatch
            options.Shadow.Color = Color.Azure
        End Sub

'#Region "Auxilliary Methods"
        Private Function CreateImages(ByVal colors As Color()) As Image()
            Dim images As Image() = New Image(colors.Length - 1) {}
            For i As Integer = 0 To colors.Length - 1
                images(i) = CreateImage(16, 16, colors(i))
            Next

            Return images
        End Function

        Private Function CreateImage(ByVal width As Integer, ByVal height As Integer, ByVal imageColor As Color) As Image
            Dim image As Bitmap = New Bitmap(width, height)
            Using gfx As Graphics = Graphics.FromImage(image)
                Using brush As SolidBrush = New SolidBrush(imageColor)
                    gfx.FillRectangle(brush, 0, 0, width, height)
                End Using
            End Using

            Return CType(image, Image)
        End Function

        Private Shared Function GetIndexByName(ByVal name As Object) As Integer
            If name IsNot Nothing Then
                For i As Integer = 0 To seriesArguments.Length - 1
                    If Equals(seriesArguments(i), name.ToString()) Then Return i
                Next
            End If

            Return -1
        End Function
'#End Region
    End Class
End Namespace
